### PR TITLE
docs: add info about PV garbage collector

### PR DESCRIPTION
**Pull request by anupriya0703**
Signed-off-by: anupriya0703 <anupriya.gupta@mayadata.io>

### CODE REVIEW

File: quickstart/faqs.md [hunk 1]
```diff
@@ -74,3 +74,11 @@ Mayastor, as any other solution leveraging TCP for network transport, may suffer
...
+
+
+## What happens when a PV with reclaim policy set to `retain` is deleted?
+
+When the reclaim policy is set to 'retain`, the underlying volume remains intact even after deletion of the PV, which leaves the volumes unusable. In this scenario, the external provisioner does not call the delete volume API. To overcome this, Mayastor deploys a PV garbage collector as part of the CSI controller. The garbage collector deletes the underlying volume object associated with the deleted PV with reclaim policy set to retain.
```
Comment (avishnu) [inline quickstart/faqs.md]: Start with a short description (one-line definition) of "retain" first. Then we can continue with "When the reclaim policy..."

File: quickstart/faqs.md [hunk 2]
```diff
@@ -74,3 +74,11 @@ Mayastor, as any other solution leveraging TCP for network transport, may suffer
...
+
+
+## What happens when a PV with reclaim policy set to `retain` is deleted?
+
+When the reclaim policy is set to 'retain`, the underlying volume remains intact even after deletion of the PV, which leaves the volumes unusable. In this scenario, the external provisioner does not call the delete volume API. To overcome this, Mayastor deploys a PV garbage collector as part of the CSI controller. The garbage collector deletes the underlying volume object associated with the deleted PV with reclaim policy set to retain. 
+
+  ### How does the PV garbage collector work? 
+    A watcher object is deployed to identify the underlying volume that is associated with the deleted PV (with reclaim policy 'retain'). When a PV is deleted, an event (generated by the watcher object) is received by the PV garbage collector. After which, the garbage collector initiates the delete operation on the Mayastor volumes using the Mayastor rest API.
```
Comment (avishnu) [inline quickstart/faqs.md]: Is the event generated by the watcher object? @abhilashshetty04 please confirm.

Comment (abhilashshetty04) [reply quickstart/faqs.md]: @anupriya0703 . watcher object will receive events from kube api server . You can use this .. Watcher in PV Garbage collector watches for changes in PV resource continuously. When a PV is deleted, an event (generated by the kube api server) is received by the watcher object. We do some validations on the PV object received from the event. After which, the garbage collector initiates the delete operation on the Mayastor volumes using the Mayastor rest API.

Comment (anupriya0703) [reply quickstart/faqs.md]: Thanks. Have updated the text.

**Review by abhilashshetty04 — APPROVED** [review]

File: quickstart/faqs.md [hunk 3]
```diff
@@ -81,4 +81,4 @@ Mayastor has been designed so as to be able to leverage the peformance capabilit
...
 
   ### How does the PV garbage collector work? 
-    A watcher object is deployed to identify the underlying volume that is associated with the deleted PV (with reclaim policy 'retain'). When a PV is deleted, an event (generated by the watcher object) is received by the PV garbage collector. After which, the garbage collector initiates the delete operation on the Mayastor volumes using the Mayastor rest API. 
\ No newline at end of file
+    A watcher object, deployed with PV Garbage, watches for PV resources continuously. When a PV is deleted, an event (generated by the kube api server) is received by the watcher object. These events are then validated. After which, the garbage collector initiates the delete operation on the Mayastor volumes using the Mayastor rest API.
```
Comment (avishnu) [inline quickstart/faqs.md]: The PV garbage collector deploys a watcher component, which subscribes to the Kubernetes Persistent Volume deletion events. When a PV is deleted, an event is generated by the Kubernetes API server and is received by this component. Upon a successful validation of this event, the garbage collector deletes the corresponding Mayastor volume objects.

@abhilashshetty04 is the above statement correct?

File: quickstart/faqs.md [hunk 4]
```diff
@@ -74,3 +74,11 @@ Mayastor, as any other solution leveraging TCP for network transport, may suffer
...
+
+
+## What happens when a PV with reclaim policy set to `retain` is deleted?
+
+The Retain reclaim policy allows for manual reclamation of the resource. When the reclaim policy is set to 'retain`, the underlying volume remains intact even after deletion of the PV, which leaves the volumes unusable. In this scenario, the external provisioner does not call the delete volume API. To overcome this, Mayastor deploys a PV garbage collector as part of the CSI controller. The garbage collector deletes the underlying volume object associated with the deleted PV with reclaim policy set to retain.
```
Comment (avishnu) [inline quickstart/faqs.md:81]: @anupriya0703 , I think the "Retain" reclaim policy has a different functionality. I think the paragraph can be re-phrased as:
In Kubernetes, when a PVC is created with reclaim policy set to 'Retain', the PV bound to this PVC is not deleted even if the PVC is deleted. One can manually delete the PV by issuing the command "kubectl delete pv <pv-name>", however the underlying storage resources could be left behind as the CSI volume provisioner (external provisioner) is not aware of this. To resolve this issue of dangling storage objects, Mayastor has introduced a PV garbage collector. This PV garbage collector is deployed as a part of the Mayastor CSI controller-plugin.

Comment (avishnu) [reply quickstart/faqs.md:81]: @abhilashshetty04 PTAL at the correctness of the above content.

Comment (abhilashshetty04) [reply quickstart/faqs.md:81]: @avishnu . This is accurate. We can use this @anupriya0703 .

**Comment by anupriya0703** [comment]
This section has been updated as part of https://github.com/openebs/mayastor-docs/pull/97

**Pull request closed without merging**